client/test/unit/specs/components/JobList.spec.js

```diff
@@ -204,6 +205,133 @@ describe('Unit | Component | JobList.vue', () => {
+              title: 'Tech Lead mission 2',
+              staffing_needed_from: '2017-10-02',
+            },
+            project: {
+              id: 123456,
+              status: 'mission_signed',
+              name: 'SCLOU - Cloud computing : enjeux, architecture et gouvernance du IaaS, CaaS, PaaS INTER 2017',
+              customer: {
+                name: 'La Poste - Courrier',
+              },
+              duration: '10 mois',
+              location: 'OCTO',
+              business_contact: {
+                nickname: 'ABC',
+              },
+              mission_director: {
+                nickname: 'XYZ',
+              },
+            },
+          },
+          {
+            id: 3,
+            activity: {
+              title: 'Tech Lead mission 3',
+              staffing_needed_from: '2017-10-03',
+            },
+            project: {
+              id: 123456,
+              status: 'proposal_in_progress',
+              name: 'SCLOU - Cloud computing : enjeux, architecture et gouvernance du IaaS, CaaS, PaaS INTER 2017',
+              customer: {
+                name: 'La Poste - Courrier',
+              },
+              duration: '10 mois',
+              location: 'OCTO',
+              business_contact: {
+                nickname: 'ABC',
+              },
+              mission_director: {
+                nickname: 'XYZ',
+              },
+            },
+          },
+          {
+            id: 4,
+            activity: {
+              title: 'Tech Lead mission 4',
+              staffing_needed_from: '2017-10-04',
+            },
+            project: {
+              id: 123456,
+              status: 'mission_signed',
+              name: 'SCLOU - Cloud computing : enjeux, architecture et gouvernance du IaaS, CaaS, PaaS INTER 2017',
+              customer: {
+                name: 'La Poste - Courrier',
+              },
+              duration: '10 mois',
+              location: 'OCTO',
+              business_contact: {
+                nickname: 'ABC',
+              },
+              mission_director: {
+                nickname: 'XYZ',
+              },
+            },
+          },
+        ];
+        expectedJobsWhenSortedByStatusAndStaffingNeededDate = [
+          jobs[3],
+          jobs[1],
+          jobs[2],
+          jobs[0],
+        ];
+
+        sinon.stub(jobsSorter, 'sort').returns(expectedJobsWhenSortedByStatusAndStaffingNeededDate);
+        sinon.stub(jobsApi, 'fetchAll').resolves(jobs);
+
+        // when
+        component = new Constructor().$mount();
+      });
+
+      afterEach(() => {
+        jobsSorter.sort.restore();
+        jobsApi.fetchAll.restore();
+      });
+
+      it('should sort the mission jobs by status and by staffing needed date', () => Vue.nextTick().then(() => Vue.nextTick().then(() => {
+        const jobTitles = component.$el.querySelectorAll('.job__title');
+        expect(jobTitles[0].textContent).to.equal('Tech Lead mission 4');
```

Comment:
peut-on nommer les missions en mode "Very old mission" "Mission for today" "yesterday mission"... bref revoir le nommage des missions pour une meilleure compréhension du texte.

Pourquoi mission 4 avant 2 puis 3 puis 1 ?
;-)